### PR TITLE
feat(web): detail-page for template & create template

### PR DESCRIPTION
**Pull request by huxiangtao**
**What this PR does / why we need it**:

create stage template
update stage template
delete stage template
stage template detail
validate template form

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @qiuxiaolian @caosm 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: web/src/components/stageTemplate/add/Form.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+@observer
+export default class StageTemplateForm extends React.Component {
+  constructor(props) {
+    super(props);
+    const {
+      match: { params },
+    } = props;
+    this.update = !!_.get(params, 'templateName');
+    if (this.update) {
+      props.stageTemplate.getTemplate(params.templateName);
```
Comment (caosm) [inline web/src/components/stageTemplate/add/Form.js]: put effect into DidMount

File: web/src/components/stageTemplate/add/Form.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+          ],
+        },
+      },
+    };
+    return _.assign(defaultSpec, specData);
+  };
+
+  initFormValue = () => {
+    const templateInfo = toJS(this.props.stageTemplate.template);
+    return this.mapRequestFormToInitForm(templateInfo);
```
Comment (caosm) [inline web/src/components/stageTemplate/add/Form.js]: map? or merge?

File: web/src/components/stageTemplate/add/Form.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+@inject('stageTemplate')
+@observer
+export default class StageTemplateForm extends React.Component {
+  constructor(props) {
+    super(props);
+    const {
+      match: { params },
+    } = props;
+    this.update = !!_.get(params, 'templateName');
```
Comment (caosm) [inline web/src/components/stageTemplate/add/Form.js]: `this.update` is not necessary

File: web/src/components/stageTemplate/add/Form.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+      stageTemplate,
+      match: { params },
+    } = this.props;
+    const submitData = this.generateData(values);
+    if (this.update) {
+      stageTemplate.updateStageTemplate(submitData, params.templateName, () => {
+        this.props.history.replace('/stageTemplate');
+      });
+    }
+    stageTemplate.createStageTemplate(submitData, () => {
```
Comment (caosm) [inline web/src/components/stageTemplate/add/Form.js]: else?

File: web/src/components/stageTemplate/add/FormContent.js [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+
+const { TextArea } = Input;
+const InputField = MakeField(Input);
+const TextareaField = MakeField(TextArea);
+const SelectField = MakeField(SelectSourceType);
+const FormItem = Form.Item;
+const Fragment = React.Fragment;
+const FormContent = props => {
+  const { setFieldValue, handleCancle, handleSubmit, update, values } = props;
+  return (
```
Comment (caosm) [inline web/src/components/stageTemplate/add/FormContent.js]: too long, split it into sub-components & make it readable

File: web/src/less/u.stageTemplate.less [hunk 6]
```diff
@@ -0,0 +1,35 @@
+.stagetemplate-form {
```
Comment (caosm) [inline web/src/less/u.stageTemplate.less]: use css-module as possibly

File: web/src/components/stageTemplate/List/Item.js [hunk 7]
```diff
@@ -1,49 +1,96 @@
...
+                e.stopPropagation();
+                history.push(`/stageTemplate/add/${name}`);
+              }}
+            />,
+            <Icon
+              key="delete"
+              type="delete"
+              onClick={e => {
+                e.stopPropagation();
+                this.setState({ visible: true });
```
Comment (caosm) [inline web/src/components/stageTemplate/List/Item.js]: `PopConfirm` is better than `Modal`

File: web/src/components/stageTemplate/List/Item.js [hunk 8]
```diff
@@ -1,49 +1,96 @@
...
-};
+    history: PropTypes.object,
+    stageTemplate: PropTypes.object,
+  };
+
+  state = {
+    visible: false,
+  };
+
+  handleSubmit = name => {
```
Comment (caosm) [inline web/src/components/stageTemplate/List/Item.js]: `handleDelete`

File: web/src/components/stageTemplate/add/Form.js (hunk 4 above)
Comment (huxiangtao) [reply web/src/components/stageTemplate/add/Form.js]: Although I think if this form is an update form, after posting update data will exec browser replace logic, but put this logic in else branch is better.

File: web/src/components/stageTemplate/add/Form.js (hunk 2 above)
Comment (huxiangtao) [reply web/src/components/stageTemplate/add/Form.js]: map ,this map is meaning just like source map , Map form data and real data

File: web/src/components/stageTemplate/add/Form.js (hunk 4 above)
Comment (caosm) [reply web/src/components/stageTemplate/add/Form.js]: it's asynchronous, `create` would still work

Comment (huxiangtao) [reply web/src/components/stageTemplate/add/Form.js]: ok

**Pull request merged**